src/Store/LifecycleStoreConnector.re

```diff
@@ -26,16 +24,17 @@ let start = quit => {
     let anyModified = Buffers.anyModified(state.buffers);
     let canClose = force || !anyModified;
 
-    Isolinear.Effect.create(~name="lifecycle.quitAll", () =>
+    Isolinear.Effect.createWithDispatch(~name="lifecycle.quitAll", dispatch =>
       if (canClose) {
+        dispatch(Actions.ReallyQuitting);
         List.iter(h => h(), handlers);
```

Comment:
Before, we used this lifecycle handler to close the syntax client. But now we need an action to go through to turn the subscription 'off' so that we can close it. I added this `ReallyQuitting` action for that purpose, but maybe there is a cleaner way.